### ISSUE TITLE
`ChildTokens` returns tokens with null Parent

### ISSUE THREAD

**Issue by Rekkonnect** [dotnet/roslyn#76441]
## Version Used
Microsoft.CodeAnalysis.CS/VB 4.12.0

## Steps to Reproduce

Create a C# syntax tree with the following code:

```csharp
#define TEST

using System;

const string greetings =
#if TEST
    "Hello Test!"
#else
    "Hello World!"
#endif
    ;
Console.WriteLine(greetings);
```

Parse the syntax tree, and get the node at a position within the `const string` portion. Within that node, get its child tokens using the `.ChildTokens()` extension. Observe the result of the `Parent` property on the tokens returned from the method.

## Expected Behavior

The parent node is the node on which we invoked the extension.

## Actual Behavior

The parent node is null.

## Side-effects

- The SyntaxTree property also returns null.
- The spans of the tokens are reported starting from 0. This is because of the null syntax tree of the node.
- We cannot get the line span of disabled trivia text, because we do not have the syntax tree to map the line span. The above example deliberately contains disabled text trivia to demonstrate this.

**Comment by CyrusNajmabadi** [comment]
Could you include the repro code for the syntax operations you're doing?

**Comment by Rekkonnect** [comment]
Apparently it's on me because I was creating a new `SyntaxTokenList` with the tokens that I was receiving, and the public API does not expose any method to provide a parent node for them. As to *why* I need a `SyntaxTokenList`, I probably don't.

The `ChildTokens` itself works as expected. Closing this out.

**Comment by CyrusNajmabadi** [comment]
> Apparently it's on me because I was creating a new `SyntaxTokenList` with the tokens that I was receiving, and the public API does not expose any method to provide a parent node for them. As to *why* I need a `SyntaxTokenList`, I probably don't.
> 
> The `ChildTokens` itself works as expected. Closing this out.

You never need to set the parent of anything. It happens automatically when a piece of syntax becomes a child of any other piece of syntax.

**Comment by Rekkonnect** [comment]
Sorry, I didn't explain it properly. The underlying tokens from the ChildTokens do return the parent node correctly. But when I create a new SyntaxTokenList and retrieve the tokens via that list, the returned tokens from the list have no parent. It is overridden from the list's internal parent which is null. The SyntaxTokenList does not provide a way to create a new instance with a set parent. Neither did I find anything in SyntaxFactory.

**Comment by CyrusNajmabadi** [comment]
That's exactly what I was referring to. 

> The SyntaxTokenList does not provide a way to create a new instance with a set parent. Neither did I find anything in SyntaxFactory.

Yes. Exactly as I said. 

You never need to set the parent of anything. It happens automatically when a piece of syntax becomes a child of any other piece of syntax.
